src/components/ControlPanel.tsx

```diff
@@ -243,6 +245,7 @@ const ControlPanel = () => {
                     key="performRecover"
                     onClick={performRecover}
                     disabled={
+                        !isDeviceSelected ||
                         isMcuboot ||
                         !isJLink ||
                         !targetIsReady ||
```

Comment:
For some reason the code has a `targetDeviceKnown` I was thinking of clearing this and reset the states in the  `targetReducer` .... but I need to understand the code more so kept this PR small to avoid any regressions